src/libslic3r/GCode.cpp

```diff
@@ -3154,8 +3157,11 @@ GCode::LayerResult GCode::process_layer(
                     // ironing
                     gcode += this->extrude_infill(print,by_region_specific, true);
                 }
-                if (GCode::gcode_label_objects)
+                if (GCode::gcode_label_objects) {
                     gcode += std::string("; stop printing object ") + instance_to_print.print_object.model_object()->name + " id:" + std::to_string(instance_to_print.layer_id) + " copy " + std::to_string(instance_to_print.instance_id) + "\n";
+                    if (!m_config.use_relative_e_distances)
```

Comment:
It will not work in case if GCode::gcode_label_objects is false. As far as i understand BS has always true gcode_label_objects, but anyway from my point of view it is more robust to not relay on something.

Reply:
Is there a reason we want to reset here?
I have inserted `G92 E0` for each layer change.
That should be able to solve any potential precision issues if that's the reason you add it.

Reply:
This is a way to make absolute extrusion compatible with excluding of objects. (forcing a reset when - and only when - label objects is active and extrusion is absolute). For instance, if there is a retraction between 2 objects and one of the object is canceled, the value of E in the retraction does not make sense (because the firmware loses track of e values). Thus the goal of this is to make sure the E value is reset to zero as soon as we are between objects (this is mandatory) and as soon as an object starts (this is as a safety but might no be required).

Reply:
> It will not work in case if GCode::gcode_label_objects is false. As far as i understand BS has always true gcode_label_objects, but anyway from my point of view it is more robust to not relay on something.

This is the point because if GCode::gcode_label_objects is false, we do not need this safety net. This is only to help exclusion of objects in a absolute extrusion context

Reply:
> 

Awesome!